.github/workflows/ci.yaml

```diff
@@ -20,429 +20,48 @@ on:
-          chmod a+x bin/regctl
-          echo "$(pwd)/bin" >> $GITHUB_PATH
-      - name: Set environment variables
-        id: vars
-        run: |
-          COMMIT_SHORT_SHA=${GITHUB_SHA:0:8}
-          LOWERCASE_REPO_OWNER=$(echo "${GITHUB_REPOSITORY_OWNER}" | awk '{print tolower($0)}')
-          echo "OPERATOR_VERSION=${COMMIT_SHORT_SHA}" >> $GITHUB_ENV
-          echo "OPERATOR_IMAGE=ghcr.io/${LOWERCASE_REPO_OWNER}/gpu-operator" >> $GITHUB_ENV
-      - name: Retag gpu-operator
-        run: |
-          regctl registry login ghcr.io -u $GITHUB_ACTOR -p ${{ secrets.GITHUB_TOKEN }}
-          regctl image copy ${OPERATOR_IMAGE}:${OPERATOR_VERSION} ${OPERATOR_IMAGE}:main-latest
-
-  push-gpu-operator-bundle-image:
-    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }} # Runs only if the event is a push to the main branch
-    needs: [release-latest-gpu-operator-image]
-    runs-on: linux-amd64-cpu4
-    steps:
-      - uses: actions/checkout@v5
-        name: Check out code
-      - name: Login to GitHub Container Registry
-        uses: docker/login-action@v3
-        with:
-          registry: ghcr.io
-          username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}
-      - name: Build bundle-image
-        env:
-          BUNDLE_IMAGE: "ghcr.io/nvidia/gpu-operator/gpu-operator-bundle:${{ github.ref_name }}-latest"
-          VERSION: ""
-          DEFAULT_CHANNEL: "stable"
-          CHANNELS: "stable"
-        run: |
-          make push-bundle-image
+  variables:
+    uses: ./.github/workflows/variables.yaml
+
+  code-scanning:
+    uses: ./.github/workflows/code-scanning.yaml
+
+  config-checks:
+    uses: ./.github/workflows/config-checks.yaml
+
+  golang-checks:
+    uses: ./.github/workflows/golang-checks.yaml
+
+  image-builds:
+    needs: [variables, config-checks, golang-checks]
+    uses: ./.github/workflows/image-builds.yaml
+    with:
+      commit_short_sha: ${{ needs.variables.outputs.commit_short_sha }}
+      label_image_source: ${{ needs.variables.outputs.label_image_source }}
+      push_on_build: ${{ needs.variables.outputs.push_on_build }}
+      operator_image_base: ${{ needs.variables.outputs.operator_image_base }}
+
+  e2e-tests:
+    needs: [variables, image-builds]
+    uses: ./.github/workflows/e2e-tests.yaml
+    with:
+      operator_image: ${{ needs.variables.outputs.operator_image_base }}
+      operator_version: ${{ needs.variables.outputs.operator_version }}
+    secrets: inherit
+
+  release:
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+    needs: [variables, e2e-tests]
+    uses: ./.github/workflows/release.yaml
+    with:
+      commit_short_sha: ${{ needs.variables.outputs.commit_short_sha }}
+      operator_version: ${{ needs.variables.outputs.operator_version }}
+      operator_image_base: ${{ needs.variables.outputs.operator_image_base }}
```

Comment:
Same comment as above -- is this really needed? Can we move these variables as outputs of the `variables` workflow itself?